### ISSUE TITLE
Display stronger warning for certificates with SHA-1 certificates

### ISSUE THREAD

**Issue by dcooper16** [drwetter/testssl.sh#358]
At the moment, testssl.sh displays a warning (pr_svrty_mediumln) if the server's certificate is signed using sha1WithRSAEncryption. I think this should be changed to pr_svrty_highln, since certificates with SHA-1 signatures may already create problems.

For the [Chrome browser](http://blog.chromium.org/2014/09/gradually-sunsetting-sha-1.html):

> 
> Sites with end-entity certificates that expire between 1 January 2016 and 31 December 2016 (inclusive), and which include a SHA-1-based signature as part of the certificate chain, will be treated as “secure, but with minor errors”.
> 
> Sites with end-entity certificates that expire on or after 1 January 2017, and which include a SHA-1-based signature as part of the certificate chain, will be treated as “affirmatively insecure”. Subresources from such domain will be treated as “active mixed content”.
> 
> The current visual display for “affirmatively insecure” is a lock with a red X, and a red strike-through text treatment in the URL scheme.

For [Microsoft browsers](https://blogs.windows.com/msedgedev/2016/04/29/sha1-deprecation-roadmap/):
> Starting with the Windows 10 Anniversary Update, Microsoft Edge and Internet Explorer will no longer consider websites protected with a SHA-1 certificate as secure and will **remove the address bar lock icon** for these sites. These sites will continue to work, but will not be considered secure. This change will be in upcoming Windows Insider Preview builds soon, and will be deployed broadly this summer. In February 2017, both Microsoft Edge and Internet Explorer will block SHA-1 signed TLS certificates.

For [Mozilla](https://blog.mozilla.org/security/2015/10/20/continuing-to-phase-out-sha-1-certificates/) the rule is:
> 1. Show the “Untrusted Connection” error whenever a SHA-1 certificate issued after January 1, 2016, is encountered in Firefox
> 2. Show the “Untrusted Connection” error whenever a SHA-1 certificate is encountered in Firefox after January 1, 2017

> We are re-evaluating when we should start rejecting all SHA-1 SSL certificates (regardless of when they were issued).  As we said before, the current plan is to make this change on January 1, 2017.  However, in light of recent attacks on SHA-1, we are also considering the feasibility of having a cut-off date as early as July 1, 2016.

**Comment by drwetter** [comment]
Hi @dcooper16,

thx first.

testssl.sh is not  looking at the whole thing from the browsers perspective. It tries to determine risks/threats.

Giving that the fuzz which Google created wrt SHA1 was IMO too much (in fact a year after their marketing went mad telling everybody the world will explode if SHA1 won't be implemented soon, their CA was still on SHA1). 

Now even later looking at the crypto SHA1 certificates do not pose a high risk. That's why I am reluctant to change that.

Admittedly that stance doesn't help web site owners if they use "testssl.sh" to check their site if a more or less big warning will be presented to the user.

Bottom line: If somebody finds an elegant way 
```
if application is http; then 
      if  (check (going through all the rules from browser vendors which change over time) is bad ); then 
           pr_TBD "you will have a problem with browsers visiting your site"
   fi
fi
``` 

that would be ok.

However looking at the warning rules and how the certificate chains  in the internet have been changed in the past half year, this will be an issue which needs to be addressed in 7 months.

Cheers, Dirk

**Comment by chrcoluk** [comment]
I think it should be a big warning on https protocol as browsers wont support it.

But on other protocols like smtp maybe just a moderate warning.

As you say sha1 is secure enough, but anyone using a sha1 certificate on their site needs to change it quite quickly due to google and mozilla changes incoming.

**Comment by drwetter** [comment]
I did put a warning in there -- if it's HTTP. The severity of the finding is the same but there's a usabilty warning in red.

**Comment by drwetter** [comment]
https://github.com/drwetter/testssl.sh/commit/f0132dcb7ff120fdb2b095155e701bda3dbbf3ff
